project/api/models.py

```diff
@@ -131,7 +131,36 @@ class CachedData(models.Model):
     event_type = models.PositiveIntegerField(choices=EVENT_TYPES)
 
     def save(self, *args, **kwargs):  # pylint: disable=W0221
-        self.rank = self.wikidata_id * randint(0, 10)  # TODO: implement #8
+        url = "https://query.wikidata.org/sparql"
+        query = """
+        SELECT ?item ?outcoming ?sitelinks ?incoming WHERE {{
+            BIND(wd:Q{wid} AS ?item)
+            ?item wikibase:statements ?outcoming.
+            ?item wikibase:sitelinks ?sitelinks.
+            {{
+                SELECT (COUNT(?s) AS ?incoming) ?item WHERE {{
+                ?s ?p ?item.
+                _:b3 wikibase:directClaim ?p.
```

Comment:
What does _:b3 mean here?

Reply:
It's a [blank node](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#QSynBlankNodes);`[]` is the shorthand syntax for declaring one. I think I had that originally but changing some things graphically in the wd query builder changed it.